pulsar-client-cpp/lib/ClientImpl.cc

```diff
@@ -588,6 +588,30 @@ uint64_t ClientImpl::newRequestId() {
     return requestIdGenerator_++;
 }
 
+uint64_t ClientImpl::getNumberOfProducers() {
+    Lock lock(mutex_);
+    uint64_t numberOfAliveProducers = 0;
+    for (const auto& producer : producers_) {
+        const auto& producerImpl = producer.lock();
+        if (producerImpl && producerImpl->isConnected()) {
+            numberOfAliveProducers++;
+        }
+    }
+    return numberOfAliveProducers;
+}
+
+uint64_t ClientImpl::getNumberOfConsumers() {
+    Lock lock(mutex_);
+    uint64_t numberOfAliveConsumers = 0;
+    for (const auto& consumer : consumers_) {
+        const auto consumerImpl = consumer.lock();
+        if (consumerImpl) {
+            if (consumerImpl->isConnected()) numberOfAliveConsumers++;
```

Comment:
You can merge these two conditions to `if (consumerImpl && consumerImpl->isConnected())`.